src/app/workspace-panel/workspace-panel.component.ts

```diff
@@ -121,13 +117,162 @@ export class WorkspacePanelComponent implements OnInit {
     this.active = this.initalConfig.active;
     this.components = this.initalConfig.components;
     this.showComponent(this.initalConfig.activeComponentId || this.components[0].id);
+    this.dragDropService.componentDroppedInsidePanel.subscribe(config => {
+
+      if (this.panelId === config.panel) {
+        console.log('component dropped inside panel', this.panelId);
+        this.addComponent({
+          id: config.component.id,
+          header: config.component.header,
+          type: config.component.type,
+        });
+      } else if (this.panelId === config.previousPanel) {
+        this.destroyComponent(config.component.id);
+      }
+
+    });
+
+    this.dragDropService.componentDroppedOutsidePanel.subscribe(config => {
+
+      if (this.panelId === config.panelId) {
+        this.destroyComponent(config.component.id);
+      }
+    });
+  }
+
+  handleHeaderMouseDown($event, headerIndex) {
+
+    $event.stopPropagation();
+    $event.preventDefault();
+    this.dragDropService.setDragStart($event.clientX , $event.clientY - this.workspaceDimensions.top, $event.offsetX, $event.offsetY, $event.target);
+    let headerEl = $event.target;
+    let firstTime = true;
+    this.mouseMoveSub = this.mouseMoveObs.subscribe(event => {
+
+      if (firstTime) {
+        this.handleHeaderDragStart($event);
```

Comment:
We only want to execute the logic inside the if statement once. Trying to think of a better way. Should probably use 'first' or some other rxjs operator on the observable.